### PR TITLE
Feature/misc add few updates

### PR DESCRIPTION
**Pull request by mouhajer-ch**
Resolves part of [ROE-853](https://companieshouse.atlassian.net/browse/ROE-853)

- ROE-853 Remove partial data options for BOs
- Add safe type checking for possible undefined date
- Ignore no-use-before-define lint rule

### CODE REVIEW

File: src/services/overseas-entities/mapping.ts [hunk 1]
```diff
@@ -29,8 +29,8 @@ const mapBeneficialOwnersIndividual = (boIndividuals: BeneficialOwnerIndividual[
         const { date_of_birth, start_date, ...rest } = boIndividual;
         boIndividualResources.push({
             ...rest,
-            date_of_birth: convertDateToIsoDateString(date_of_birth.day, date_of_birth.month, date_of_birth.year),
-            start_date: convertDateToIsoDateString(start_date.day, start_date.month, start_date.year)
+            date_of_birth: convertDateToIsoDateString(date_of_birth?.day, date_of_birth?.month, date_of_birth?.year),
```
Comment (mwestacott) [inline src/services/overseas-entities/mapping.ts:32]: Just wondering why we're doing this - aren't we in the process of adding validation to ensure that all date fields will be populated, so incomplete dates should never make it into the SDK?

Comment (mouhajer-ch) [reply src/services/overseas-entities/mapping.ts:32]: Indeed, but as long as those field are set to be optional better add a safe check and avoid this [here](https://companieshouse.atlassian.net/browse/ROE-914?focusedCommentId=195717)

Comment (mwestacott) [reply src/services/overseas-entities/mapping.ts:32]: So we'll revert this change later on, once fields are no longer optional?

Comment (mouhajer-ch) [reply src/services/overseas-entities/mapping.ts:32]: Yep, probably when we start the API validation we set optionals/required accordantly and remove the blank date checking

Comment (mwestacott) [reply src/services/overseas-entities/mapping.ts:32]: Still not clear sorry. Surely the optionality of these fields in the SDK will be driven by the level of validation in the web app (not API) - so overseas-entities-web - if the web app dictates that the day, month, year fields should have values in (via validation) then they should NOT be marked optional here.

Comment (mwestacott) [reply src/services/overseas-entities/mapping.ts:32]: Looks like definitions of optionals in the SDK model means this must stay like this (for now).

**Review by mwestacott — APPROVED** [review]

**Pull request merged**